controllers/v1/sprint_highlight.go

```diff
@@ -26,38 +28,90 @@ func (ctrl SprintHighlightController) Add(c *gin.Context) {
 	userID, _ := c.Get("userID")
 	sprintID := c.Param("sprintID")
 	retroID := c.Param("retroID")
+	feedbackData := serializers.RetrospectiveFeedbackCreateSerializer{}
+
+	if err := c.BindJSON(&feedbackData); err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"message": "Invalid request data", "error": err.Error()})
+		return
+	}
 
 	if !ctrl.PermissionService.UserCanEditSprint(retroID, sprintID, userID.(uint)) {
 		c.AbortWithStatusJSON(http.StatusForbidden, gin.H{})
 		return
 	}
 
-	c.JSON(http.StatusOK, gin.H{})
+	response, err := ctrl.RetrospectiveFeedbackService.Add(
+		userID.(uint),
+		sprintID,
+		retroID,
+		models.HighlightType,
+		&feedbackData)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{
+			"message": "Failed to create highlight",
+			"error":   err.Error()})
+		return
+	}
+
+	c.JSON(http.StatusCreated, response)
 }
 
 // List highlights associated to sprint
 func (ctrl SprintHighlightController) List(c *gin.Context) {
 	userID, _ := c.Get("userID")
 	sprintID := c.Param("sprintID")
 	retroID := c.Param("retroID")
+
 	if !ctrl.PermissionService.UserCanAccessSprint(retroID, sprintID, userID.(uint)) {
 		c.AbortWithStatusJSON(http.StatusForbidden, gin.H{})
 		return
 	}
-	c.JSON(http.StatusOK, gin.H{})
+
+	response, err := ctrl.RetrospectiveFeedbackService.List(
+		userID.(uint),
+		sprintID,
+		retroID,
+		models.HighlightType)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{
+			"message": "Failed to fetch highlights",
+			"error":   err.Error()})
+		return
+	}
+
+	c.JSON(http.StatusOK, response)
 }
 
 // Update highlight associated to a sprint
 func (ctrl SprintHighlightController) Update(c *gin.Context) {
 	userID, _ := c.Get("userID")
 	sprintID := c.Param("sprintID")
 	retroID := c.Param("retroID")
-	//highlightID := c.Param("highlightID")
+	highlightID := c.Param("highlightID")
+	feedbackData := serializers.RetrospectiveFeedbackUpdateSerializer{}
+
+	if err := c.BindJSON(&feedbackData); err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"message": "Invalid request data", "error": err.Error()})
+		return
+	}
 
 	if !ctrl.PermissionService.UserCanEditSprint(retroID, sprintID, userID.(uint)) {
 		c.AbortWithStatusJSON(http.StatusForbidden, gin.H{})
 		return
 	}
 
-	c.JSON(http.StatusOK, gin.H{})
+	response, err := ctrl.RetrospectiveFeedbackService.Update(
+		userID.(uint),
+		sprintID,
+		retroID,
+		highlightID,
+		&feedbackData)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{
+			"message": "Failed to update highlight",
+			"error":   err.Error()})
+		return
+	}
+
```

Comment:
Let's add an audit trail here and in other update actions

Reply:
done